File: leecode/code/102/102.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<TreeNode> record = new ArrayList();
        List<List<Integer>> res = new ArrayList();
        record.add(root);
        List<Integer> k = new ArrayList();
        int count = 1;
        while (record.size() != 0 && root != null) {
            TreeNode temp = record.remove(0);
            k.add(temp.val);
            count--;
            if (temp.left != null) {
                record.add(temp.left);
            }
            if (temp.right != null) {
                record.add(temp.right);
            }
            if (count == 0) {
                res.add(k);
                count = record.size();
                k = new ArrayList();
            }
        }
        return res;
    }
}

File: leecode/code/102/102.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def dfs(self, root, level, res):
        if not root: 
            return
        if len(res) == level:
            res.append([])
        res[level].append(root.val)
        if root.left != None:
            self.dfs(root.left, level + 1, res)
        if root.right != None:
            self.dfs(root.right, level + 1, res)



    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        res = []
        self.dfs(root, 0, res)
        return res

